diff --git a/README.md b/README.md
index e32f491..def12f4 100644
--- a/README.md
+++ b/README.md
@@ -22,7 +22,7 @@ Simple Neovim Config
 </details>
 
 ## :clipboard: Requirements
-- Neovim >= 0.9.5
+- Neovim >= 0.10
 - Linux or macOS
 
 ## :package: Installation
@@ -33,10 +33,10 @@ git clone "https://github.com/notfirefox/nvim-config.git" ~/.config/nvim
 
 ###### macOS
 ```sh
-brew install fd make neovim ripgrep tree-sitter
+brew install fd neovim ripgrep tree-sitter
 ```
 
 ###### Ubuntu
 ```sh
-sudo apt install fd-find make neovim ripgrep tree-sitter-cli
+sudo apt install fd-find neovim ripgrep tree-sitter-cli
 ```
diff --git a/init.lua b/init.lua
index 0b906aa..b580c76 100644
--- a/init.lua
+++ b/init.lua
@@ -1,4 +1,3 @@
-vim.uv = vim.uv or vim.loop -- v0.9.5 compatibility
 require("core.config")
 require("core.event")
 require("core.plugin")
diff --git a/lazy-lock.json b/lazy-lock.json
index e239284..9f2054e 100644
--- a/lazy-lock.json
+++ b/lazy-lock.json
@@ -1,7 +1,5 @@
 {
-  "LuaSnip": { "branch": "master", "commit": "c9b9a22904c97d0eb69ccb9bab76037838326817" },
   "cmp-nvim-lsp": { "branch": "main", "commit": "99290b3ec1322070bcfb9e846450a46f6efa50f0" },
-  "cmp_luasnip": { "branch": "master", "commit": "98d9cb5c2c38532bd9bdb481067b20fea8f32e90" },
   "gruvbox.nvim": { "branch": "main", "commit": "c6f42890551b4827253387e93b035568826a9cb7" },
   "lazy.nvim": { "branch": "main", "commit": "6c3bda4aca61a13a9c63f1c1d1b16b9d3be90d7a" },
   "nvim-autopairs": { "branch": "master", "commit": "6522027785b305269fa17088395dfc0f456cedd2" },
diff --git a/lua/core/config.lua b/lua/core/config.lua
index 8007df7..bce3bc6 100644
--- a/lua/core/config.lua
+++ b/lua/core/config.lua
@@ -26,7 +26,7 @@ vim.g.loaded_python3_provider = 0
 vim.g.loaded_ruby_provider = 0
 
 -- set leader key
-vim.g.mapleader = " "
+vim.g.mapleader = vim.keycode("<space>")
 
 -- include loading status
 function CustomRulerFormat()
diff --git a/lua/core/event.lua b/lua/core/event.lua
index f25af41..e9aa1aa 100644
--- a/lua/core/event.lua
+++ b/lua/core/event.lua
@@ -44,47 +44,30 @@ vim.api.nvim_create_autocmd("FileType", {
 })
 
 -- update loading status
-local function update_loading_status(done)
-    local spinner = {
-        "⠋",
-        "⠙",
-        "⠹",
-        "⠸",
-        "⠼",
-        "⠴",
-        "⠦",
-        "⠧",
-        "⠇",
-        "⠏",
-    }
-    if done then
-        vim.g.loading_status = nil
-    else
-        vim.g.loading_status =
-            spinner[math.floor(vim.uv.hrtime() / (1e6 * 80)) % #spinner + 1]
-    end
-    vim.cmd.redrawstatus()
-end
-
--- v0.9.5 compatibility
-if vim.fn.has("nvim-0.10") == 0 then
-    vim.api.nvim_create_autocmd("User", {
-        pattern = "LspProgressUpdate",
-        group = vim.api.nvim_create_augroup("lsp_progress", { clear = true }),
-        callback = function()
-            local done = vim.lsp.util.get_progress_messages()[1].done
-            update_loading_status(done)
-        end,
-    })
-else
-    vim.api.nvim_create_autocmd("LspProgress", {
-        group = vim.api.nvim_create_augroup("lsp_progress", { clear = true }),
-        callback = function(ev)
-            local done = ev.data.params.value.kind == "end"
-            update_loading_status(done)
-        end,
-    })
-end
+vim.api.nvim_create_autocmd("LspProgress", {
+    group = vim.api.nvim_create_augroup("lsp_progress", { clear = true }),
+    callback = function(ev)
+        local spinner = {
+            "⠋",
+            "⠙",
+            "⠹",
+            "⠸",
+            "⠼",
+            "⠴",
+            "⠦",
+            "⠧",
+            "⠇",
+            "⠏",
+        }
+        if ev.data.params.value.kind == "end" then
+            vim.g.loading_status = nil
+        else
+            vim.g.loading_status =
+                spinner[math.floor(vim.uv.hrtime() / (1e6 * 80)) % #spinner + 1]
+        end
+        vim.cmd.redrawstatus()
+    end,
+})
 
 -- configure language server
 vim.api.nvim_create_autocmd("LspAttach", {
@@ -96,23 +79,24 @@ vim.api.nvim_create_autocmd("LspAttach", {
         end
 
         local bind = vim.keymap.set
+        local methods = vim.lsp.protocol.Methods
         local opts = { buffer = args.buf }
         local utils = require("core.utils")
 
-        if client:supports_method("textDocument/codeAction") then
+        if client:supports_method(methods.textDocument_codeAction) then
             bind("n", "<leader>ca", vim.lsp.buf.code_action, opts)
         end
 
-        if client:supports_method("textDocument/hover") then
+        if client:supports_method(methods.textDocument_hover) then
             bind("n", "K", vim.lsp.buf.hover, opts)
         end
 
-        if client:supports_method("textDocument/formatting") then
+        if client:supports_method(methods.textDocument_formatting) then
             utils.add_formatting_autocmd(args.buf, client.id)
             utils.create_clang_format()
         end
 
-        if client:supports_method("textDocument/rename") then
+        if client:supports_method(methods.textDocument_rename) then
             bind("n", "<leader>rn", vim.lsp.buf.rename, opts)
         end
     end,
diff --git a/lua/plugins/completion.lua b/lua/plugins/completion.lua
index 4e8901b..a1e98a4 100644
--- a/lua/plugins/completion.lua
+++ b/lua/plugins/completion.lua
@@ -1,17 +1,8 @@
 return {
     "hrsh7th/nvim-cmp",
-    dependencies = {
-        {
-            "L3MON4D3/LuaSnip",
-            build = "gmake install_jsregexp",
-            tag = "v2.*",
-        },
-        "hrsh7th/cmp-nvim-lsp",
-        "saadparwaiz1/cmp_luasnip",
-    },
+    dependencies = { "hrsh7th/cmp-nvim-lsp" },
     config = function()
         -- variables
-        local luasnip = require("luasnip")
         local cmp_autopairs = require("nvim-autopairs.completion.cmp")
         local cmp = require("cmp")
 
@@ -42,11 +33,6 @@ return {
                     return vim_item
                 end,
             },
-            snippet = {
-                expand = function(args)
-                    luasnip.lsp_expand(args.body)
-                end,
-            },
             mapping = cmp.mapping.preset.insert({
                 ["<C-Space>"] = cmp.mapping.complete(),
                 ["<CR>"] = cmp.mapping.confirm({
@@ -54,15 +40,15 @@ return {
                     select = true,
                 }),
                 ["<Tab>"] = cmp.mapping(function(fallback)
-                    if luasnip.expand_or_jumpable() then
-                        luasnip.expand_or_jump()
+                    if vim.snippet.active({ direction = 1 }) then
+                        vim.snippet.jump(1)
                     else
                         fallback()
                     end
                 end, { "i", "s" }),
                 ["<S-Tab>"] = cmp.mapping(function(fallback)
-                    if luasnip.jumpable(-1) then
-                        luasnip.jump(-1)
+                    if vim.snippet.active({ direction = -1 }) then
+                        vim.snippet.jump(-1)
                     else
                         fallback()
                     end
@@ -70,7 +56,6 @@ return {
             }),
             sources = {
                 { name = "nvim_lsp" },
-                { name = "luasnip" },
             },
         })
     end,
